src/cephadm/cephadm.py

```diff
@@ -7463,14 +7492,18 @@ def get_ceph_cluster_count(ctx: CephadmContext) -> int:
     return len([c for c in os.listdir(ctx.data_dir) if is_fsid(c)])
 
 
-def command_rm_cluster(ctx):
+def command_rm_cluster(ctx, internal=False):
     # type: (CephadmContext) -> None
-    if not ctx.force:
+    if not internal and not ctx.force:
         raise Error('must pass --force to proceed: '
                     'this command may destroy precious data!')
 
-    lock = FileLock(ctx, ctx.fsid)
-    lock.acquire()
+    if not ctx.fsid:
+        raise Error('must select the cluster to delete by passing --fsid to proceed')
+
+    if not internal:
```

Comment:
Having to do this makes me think that command_rm_cluster deserves to be split into (at least) two parts: and "upper half" that does the command handling  and a "lower half" that is common and does the actual work of the clean up. I recently did something similar in my PR #51738 - this may make the code flow better.

Reply:
@phlogistonjohn good idea. I split the command in two to have a clear interface/params (I tried to use the same that are passed from the cmd line) the rest are part of the context.